### PR TITLE
add docs for manually resyncing a broken replica

### PR DESCRIPTION
**Pull request by satyanash**
Given the resolutions to the following issues: #562 #604, I felt that having an explicit doc would increase visibility for the solution. Hence this PR.

### CODE REVIEW

**Comment by satyanash** [comment]
@sgotti The failures look intermittent and unrelated.

File: doc/resync_broken_replica.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Re-sync a broken replica
+
+Due to unforeseen circumstances, if you end up with an empty postgres data directory on one of your replicas but your _master_ is intact, you will receive a message that looks something like this:
+
+```
+different local dbUID but init mode is none, this shouldn't happen. Something bad happened to the keeper data. Check that keeper data is on a persistent volume and that the keeper state files weren't removed
+```
+
+This is because all keepers _require_ persistent storage and will not auto "re-sync" just because they find an empty data directory while still having an entry in the clusterdata.
```
Comment (sgotti) [inline doc/resync_broken_replica.md]: I'll add a link to the `architecture.md` doc that clearly explains this.

Comment (satyanash) [reply doc/resync_broken_replica.md]: @sgotti I've added the link under the Misc. section, and also updated `architecture.md` to specify the behaviour in case of a cluster ID mismatch. Please review.

File: doc/README.md [hunk 2]
```diff
@@ -22,6 +22,7 @@ We suggest that you first read the [Stolon Architecture and Requirements](archit
 * [PostgreSQL SSL/TLS setup](ssl.md)
 * [Forcing a failover](forcefailover.md)
 * [Service Discovery](service_discovery.md)
+* [Resync a Broken Replica](resync_broken_replica.md)
```
Comment (sgotti) [inline doc/README.md]: To keep naming consistent I'll rename it to `Reset/Resync a broken keeper`

File: doc/architecture.md [hunk 3]
```diff
@@ -14,11 +14,10 @@ Stolon is composed of 3 main components
...
-
-If you're providing the keeper's uid in the command line don't start a new keeper with the same id if you're providing a different data directory (empty or populated) since you're changing data out of the stolon control causing possible data loss or strange behaviors.
+* Every keeper MUST have a different UID that can be manually provided (`--uid` option) or will be generated. After the first start the keeper id (provided or generated) is saved inside the keeper data directory. If you start two distinct keeper instances with the same UID, expect strange behaviour and/or data loss.
+* Every keeper MUST have a persistent data directory (no ephemeral volumes like k8s `emptyDir`) or you'll lose your data if all the keepers are stopped at the same time (since at restart no valid standby to failover will be available).
+* If an existing functioning keeper, finds that it's data directory is empty, it will refuse to start until the data directory is available again.
+* If an existing functioning keeper, finds that it's current Postgres Cluster ID is different from the one it previously had in the clusterdata, it will still refuse to start.
```
Comment (sgotti) [inline doc/architecture.md]: This is out of scope for the current PR. If you want to change the architecture.md doc please open a new PR.

File: doc/resync_broken_replica.md [hunk 4]
```diff
@@ -0,0 +1,18 @@
+# Re-sync a broken replica
```
Comment (sgotti) [inline doc/resync_broken_replica.md]: To keep naming consistent I'll rename it to Reset/Resync a broken keeper

File: doc/resync_broken_replica.md (hunk 1 above)
Comment (sgotti) [inline doc/resync_broken_replica.md]: This is just a case of "broken keeper", there can be many other cases so I'll make it more generic and make this a possible case (empty postgres data due to **wrong** user configuration)

File: doc/resync_broken_replica.md (hunk 4 above)
Comment (satyanash) [reply doc/resync_broken_replica.md]: @sgotti I use the term "replica" because one cannot resync _any_ keeper. The master has to be intact for this to work.

Comment (sgotti) [reply doc/resync_broken_replica.md]: What about `Reset/Resync a broken standby keeper`?

Comment (satyanash) [reply doc/resync_broken_replica.md]: Will incorporate this suggestion.
